mgc/dscr/oneSample.py

```diff
@@ -0,0 +1,110 @@
+from sklearn.metrics import euclidean_distances
+from ._utils import _CheckInputs
+import numpy as np
+import random
+from .base import DiscriminabilityTest
+from scipy._lib._util import MapWrapper
+
+
+
+class oneSample(DiscriminabilityTest):
+    r"""
+     A class that performs a one-sample test for whether the discriminability differs from random chance,
+     as described in [1]. With :math:`D_X` as the sample discriminability of :math:`X`, it tests whether:
+
+     .. math::
+        H_0: D_X = D_0
+
+     and
+
+     .. math::
+        H_A: D_X > D_0 
+    
+    where :math:`D_0` is the discriminability that would be observed by random chance.
+    
+    References
+    ----------
+    .. [#1Dscr] Eric W. Bridgeford, et al. "Optimal Decisions for Reference 
+                Pipelines and Datasets: Applications in Connectomics." Bioarxiv (2019).
+    """
+
+    def __init__(self):
+        self.stat = None
+        self.pvalue = None
+        DiscriminabilityTest.__init__(self)
+
+
+
+    def test(self, X, Y, isDist = False, remove_isolates=True, reps=1000, workers=-1):
+        r"""
+        Calculates the test statistic and p-value for Discriminability one sample test.
+
+        Parameters
+        ----------
+        X : ndarray
+
+            * An :math:`n \times d` dimensional data matrix with :math:`n` samples in :math:`d` dimensions, if flag :math:`(isDist = Flase)` 
+
+            * An :math:`n \times n` dimensional distance matrix if :math:`X` is a distance matrix. Use flag :math:`(isDist = True)`
+            
+        isDist : Boolean, optional (default: False)
```

Comment:
consider calling this `dissimilarity="precomputed"` like sklearn maybe?

Reply:
According to the R code, it was is_dist. And I want the default parameter to be "not precomputed" which may look bad.

Reply:
`is_dist` can be very easily calculated, look at other examples like `Dcorr`